internal/resources/iampolicy/scope_helper.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	iammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy"
+)
+
+type scopeType int
+
+const (
+	unknown scopeType = iota
+	organization
+	clusterGroup
+	cluster
+	workspace
+	namespace
+)
+
+func stringToScope(s string) scopeType {
+	switch s {
+	case organizationKey:
+		return organization
+	case clusterGroupKey:
+		return clusterGroup
+	case clusterKey:
+		return cluster
+	case workspaceKey:
+		return workspace
+	case namespaceKey:
+		return namespace
+	default:
+		return unknown
+	}
+}
+
+func (s scopeType) String() string {
+	return [...]string{
+		"unknown",
+		organizationKey,
+		clusterGroupKey,
+		clusterKey,
+		workspaceKey,
+		namespaceKey,
+	}[s]
+}
```

Comment:
Why do we need this ? We already have declared a scope type enum with an `unknown` value for no scope.